scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -1530,9 +1623,14 @@ public final void send(final RawData msg) {
 			error = new IllegalArgumentException(
 					"Message data must not exceed " + MAX_PLAINTEXT_FRAGMENT_LENGTH + " bytes");
 		} else {
-			serialExecutor = getSerialExecutor(msg.getInetSocketAddress());
-			if (serialExecutor == null) {
-				error = new IllegalStateException("connector's serial-executors-cache is exhausted!");
+			connection = getAliveConnection(msg.getInetSocketAddress(), !serverOnly);
+			if (connection == null) {
+				if (connectionStore.remainingCapacity() > 0 && serverOnly) {
```

Comment:
if `connectionStore.remainingCapacity() > 0` and we are not in `serverOnly` mode we raise `new IllegalStateException("connection store is exhausted!")`, is it normal ?

Reply:
If `serverOnly` is `false`, new connections are created, and so `null` is returned, if the the store is exhausted and the `put` failed.
If `serverOnly` is `true`, the `remaining` call is not required. I will remove it.